### DISCUSSION TITLE
Can we get the output of run_javascript() at backend?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1560] [Q&A] [answered]
### Question

Hi All,

I am using the ```ui.run_javascript``` to get some info like screen width. My intention is to capture this in a backend variable and then use it for my logic, let's say display a different image or some other conditions based on the screen width returned.

I tried the following but it does not work:
```python
ui.button('Click', on_click=lambda: screen_width())
label = ui.label('This is blank')

async def screen_width():
    await ui.run_javascript(f'getElement({label.id}).innerText = window.innerWidth')
    print(label.text)
```
The print command does not print the new text from the label, although on my webpage the value is updated. Seems like it's not getting transferred to the backend. I tried the value binding to a variable also, but that too, I could not make it work.

Can you please suggest me what will be the proper way to get the output of the js code in the backend python variable?

Thanks,
Anindya

**Comment by rodja** [comment, marked as answer]
Have you seen the [`get_date` demo in the docs](https://nicegui.io/documentation/run_javascript#:~:text=async%20def%20get_date()%3A%0A%20%20%20%20time%20%3D%20await%20ui.run_javascript(%27Date()%27)%0A%20%20%20%20ui.notify(f%27Browser%20time%3A%20%7Btime%7D%27))? You can simply run javascript which generates a value and assign the result to a variable.

Your code is not working because you use javascript to modify the frontend data directly. The backend does not know about this and so `label.text` stays the same.

**Reply by Anindya088** [reply]
Hi @rodja ,

Oh this is so great! So basically ```ui.javascript()``` behaves like a normal python function and can return value to a normal python variable. 
This is so cool! 

Sorry for missing out the doc, should have read more thoroughly through the examples.

Thanks,
Anindya
